apm-sniffer/config/agent.config

```diff
@@ -15,100 +15,243 @@
+plugin.http.http_params_length_threshold=${SW_PLUGIN_HTTP_HTTP_PARAMS_LENGTH_THRESHOLD:1024}
+#  When `include_http_headers` declares header names, this threshold controls the length limitation of all header values. use negative values to keep and send the complete headers. Note. this config item is added for the sake of performance. 
+plugin.http.http_headers_length_threshold=${SW_PLUGIN_HTTP_HTTP_HEADERS_LENGTH_THRESHOLD:2048}
+#  Set the header names, which should be collected by the plugin. Header name must follow `javax.servlet.http` definition. Multiple names should be split by comma.
+plugin.http.include_http_headers=${SW_PLUGIN_HTTP_INCLUDE_HTTP_HEADERS:}
+#  This config item controls that whether the Feign plugin should collect the http body of the request. 
+plugin.feign.collect_request_body=${SW_PLUGIN_FEIGN_COLLECT_REQUEST_BODY:false}
+#  When `COLLECT_REQUEST_BODY` is enabled, how many characters to keep and send to the OAP backend, use negative values to keep and send the complete body.  
+plugin.feign.filter_length_limit=${SW_PLUGIN_FEIGN_FILTER_LENGTH_LIMIT:1024}
+#  When `COLLECT_REQUEST_BODY` is enabled and content-type start with SUPPORTED_CONTENT_TYPES_PREFIX, collect the body of the request , multiple paths should be separated by `,`  
+plugin.feign.supported_content_types_prefix=${SW_PLUGIN_FEIGN_SUPPORTED_CONTENT_TYPES_PREFIX:application/json,text/}
+# If true, trace all the influxql(query and write) in InfluxDB access, default is true.
+plugin.influxdb.trace_influxql=${SW_PLUGIN_INFLUXDB_TRACE_INFLUXQL:true}
+#  Apache Dubbo consumer collect `arguments` in RPC call, use `Object#toString` to collect `arguments`. 
+plugin.dubbo.collect_consumer_arguments=${SW_PLUGIN_DUBBO_COLLECT_CONSUMER_ARGUMENTS:false}
+#  When `plugin.dubbo.collect_consumer_arguments` is `true`, Arguments of length from the front will to the OAP backend 
+plugin.dubbo.consumer_arguments_length_threshold=${SW_PLUGIN_DUBBO_CONSUMER_ARGUMENTS_LENGTH_THRESHOLD:256}
+#  Apache Dubbo provider collect `arguments` in RPC call, use `Object#toString` to collect `arguments`. 
+plugin.dubbo.collect_provider_arguments=${SW_PLUGIN_DUBBO_COLLECT_PROVIDER_ARGUMENTS:false}
+#  When `plugin.dubbo.provider_consumer_arguments` is `true`, Arguments of length from the front will to the OAP backend 
+plugin.dubbo.consumer_provider_length_threshold=${SW_PLUGIN_DUBBO_CONSUMER_PROVIDER_LENGTH_THRESHOLD:256}
+#  A list of host/port pairs to use for establishing the initial connection to the Kafka cluster. 
+plugin.kafka.bootstrap_servers=${SW_KAFKA_BOOTSTRAP_SERVERS:localhost:9092}
+#  Timeout period of reading topics from the Kafka server, the unit is second. 
+plugin.kafka.get_topic_timeout=${SW_GET_TOPIC_TIMEOUT:10}
+#  Kafka producer configuration.
+plugin.kafka.consumer_config=${SW_PLUGIN_KAFKA_CONSUMER_CONFIG:}
```

Comment:
Can't see where is using this config item?

Reply:
I think it is in `KafkaProducerManager#run`. 

@dmsolr Could you be clear about how to add this property? I mean for a hashmap type.

Reply:
It is my mistake. The proper configuration item is `plugin.kafka.producer_config`.

The format is
```
plugin.kafka.producer_config[key]=value
```
and the key list was [here](https://kafka.apache.org/23/documentation.html#producerconfigs). 
I do not try to set it in JVM Options. It seems that you can't do it.

Reply:
@kezhenxu94 Any preference to do this? This mode is hard to override through system env. We can't list them all, they are too many.

Reply:
> @kezhenxu94 Any preference to do this? This mode is hard to override through system env. We can't list them all, they are too many.

What about avoiding the key in the config key, for example `plugin.kafka.producer_config=key1=value1,key2=value2`, if we could use JSON in agent, we may use `plugin.kafka.producer_config={"key":"val"}`?

Reply:
> plugin.kafka.producer_config=key1=value1,key2=value2

I prefer to support this first in the agent initialization through another PR? @dmsolr Could you do this?

Reply:
> plugin.kafka.producer_config=key1=value1,key2=value2
It is good to me.

> @dmsolr Could you do this?
Sure.

Reply:
Please create an issue to track this.